src/transformers/trainer.py

```diff
@@ -573,8 +573,7 @@ def _training_step(
         self, model: nn.Module, inputs: Dict[str, torch.Tensor], optimizer: torch.optim.Optimizer
     ) -> float:
         model.train()
-        for k, v in inputs.items():
-            inputs[k] = v.to(self.args.device)
+        inputs = {k: v.to(self.args.device) if isinstance(v, torch.Tensor) else v for k, v in inputs.items()}
 
```

Comment:
I find this personally hard to ready.

Maybe:

```python
for k, v in inputs.items():
    if isinstance(v, torch.Tensor):
        inputs[k] = v.to(self.args.device)
```

Reply:
what do you think?

Reply:
(otherwise, LGTM – we might want to document this in the function's type signature – or not)

Reply:
Usually using list-comprehensions are faster, but the batch don't have a lot of keys so it probably doesn't matter.